client/my-sites/site-settings/podcast-cover-image-setting/index.jsx

```diff
@@ -0,0 +1,318 @@
+import { AspectRatios } from 'state/ui/editor/image-editor/constants';
+import Spinner from 'components/spinner';
+
+/**
+ * Debug
+ */
+import debugModule from 'debug';
+const debug = debugModule( 'calypso:podcast-image' );
+
+class PodcastCoverImageSetting extends PureComponent {
+	static propTypes = {
+		coverImageId: PropTypes.number,
+		coverImageUrl: PropTypes.string,
+		onRemove: PropTypes.func,
+		onSelect: PropTypes.func,
+	};
+
+	state = {
+		hasToggledModal: false,
+		isEditingCoverImage: false,
+		isModalVisible: false,
+		transientMediaId: null,
+	};
+
+	toggleModal = isModalVisible => {
+		const { isEditingCoverImage } = this.state;
+
+		this.setState( {
+			isModalVisible,
+			hasToggledModal: true,
+			isEditingCoverImage: isModalVisible ? isEditingCoverImage : false,
+		} );
+	};
+
+	showModal = () => this.toggleModal( true );
+
+	hideModal = () => this.toggleModal( false );
+
+	editSelectedMedia = value => {
+		if ( value ) {
+			this.setState( { isEditingCoverImage: true } );
+			this.props.onEditSelectedMedia();
+		} else {
+			this.hideModal();
+		}
+	};
+
+	uploadCoverImage( blob, fileName ) {
+		const { siteId, site } = this.props;
+
+		// Upload media using a manually generated ID so that we can continue
+		// to reference it within this function
+		const transientMediaId = uniqueId( 'podcast-cover-image' );
+
+		this.setState( { transientMediaId } );
+
+		const checkUploadComplete = () => {
+			// MediaStore tracks pointers from transient media to the persisted
+			// copy, so if our request is for a media which is not transient,
+			// we can assume the upload has finished.
+			const media = MediaStore.get( siteId, transientMediaId );
+			const isUploadInProgress = media && isItemBeingUploaded( media );
+			const isFailedUpload = ! media;
+
+			if ( isFailedUpload ) {
+				this.props.deleteMedia( siteId, transientMediaId );
+			} else {
+				this.props.receiveMedia( siteId, media );
+			}
+
+			if ( isUploadInProgress ) {
+				return;
+			}
+
+			MediaStore.off( 'change', checkUploadComplete );
+
+			if ( ! isFailedUpload ) {
+				debug( 'upload media', media );
+				this.props.onSelect( media.ID, media.URL );
+			}
+
+			// Remove transient image so that new image shows or if failed upload, the prior image
+			this.setState( { transientMediaId: null } );
+		};
+
+		MediaStore.on( 'change', checkUploadComplete );
+
+		MediaActions.add( site, {
+			ID: transientMediaId,
+			fileContents: blob,
+			fileName,
+		} );
+	}
+
+	setCoverImage = ( error, blob ) => {
+		if ( error || ! blob ) {
+			return;
+		}
+
+		const { siteId } = this.props;
+		const selectedItem = head( MediaLibrarySelectedStore.getAll( siteId ) );
+		if ( ! selectedItem ) {
+			return;
+		}
+
+		debug( 'selectedItem', selectedItem );
+
+		const { crop, transform } = this.props;
+		const isImageEdited = ! isEqual(
+			{
+				...crop,
+				...transform,
+			},
+			{
+				topRatio: 0,
+				leftRatio: 0,
+				widthRatio: 1,
+				heightRatio: 1,
```

Comment:
In Chrome, I've occasionally seen this check fail with (for example) `leftRatio: .003..., widthRatio: .996...`, incorrectly indicating a very slight edit to the image, which causes a new image to be uploaded.

This failure isn't reliable - I suspect that it has something to do with window and/or image size.

Reply:
I've seen similar behavior too, but haven't been able to narrow down the cause.

Reply:
This is an existing issue with the image editor.  Let's track it separately.